magento/pwa-studio
# [bug]: debounce search input

## Description

This is bad:

[![Image from Gyazo](https://i.gyazo.com/6a29283668cd2f6b8c11255050815b35.gif)](https://gyazo.com/6a29283668cd2f6b8c11255050815b35)

We had hoped to be able to cancel the query on new input but that isn't something the Apollo Client can do. Instead of cancelling, we will debounce the query function so we only run it 500ms after the last change.

## Related Issue





Closes #1225 .

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Open the search bar and type some long word really quickly. The query should only run after you complete typing.

## Screenshots / Screen Captures (if appropriate)
[![Image from Gyazo](https://i.gyazo.com/b86083fcf54d0d3b2b17adbe377ca9b6.gif)](https://gyazo.com/b86083fcf54d0d3b2b17adbe377ca9b6)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
